public.openapi.yaml

```diff
@@ -979,7 +979,7 @@ components:
             type: string
             maxLength: 63
             minLength: 1
-            pattern: ^[a-z][a-z0-9\-]*$
+            pattern: ^[a-zA-Z][a-zA-Z0-9\-]*$
```

Comment:
Per RFC 1123, first character can be a digit.  Also, the last character cannot be must be a letter or digit, not `-`.

Reply:
Updated

Reply:
Just for reference, similar utility method in IPA: https://github.com/freeipa/freeipa/blob/master/ipalib/util.py#L389

Reply:
I was also thinking that we could allow any string and not have a regex here. We get the information from IPA server  - IPA locations. So from that PoV we should support what it supports.  

I just tried to define some random locations and it allows various other chars. It looks that the IPA location name is not much restricted even though it is defined as [`DNSNameParam`](https://github.com/freeipa/freeipa/blob/master/ipaserver/plugins/location.py#L99). 

![image](https://github.com/user-attachments/assets/c9cdee42-2e72-4300-b7a3-b98316d9df97)

Reply:
> It looks that the IPA location name is not much restricted even though it is defined as [`DNSNameParam`](https://github.com/freeipa/freeipa/blob/master/ipaserver/plugins/location.py#L99).

hmmm... so there exists a real possibility that a customer could have non-conformant location data, which would cause registration failure.

I am OK with completely relaxing the location validation in our service.  Bad data might (or might not, depending how their DNS is configured) be an operational issue for the IPA server.  But I think we should not make the bad data prevent registration with idmsvc.

However, let's bring the issue up with the IPA team before committing.  They might agree it's a bug and decide to properly restrict the field.  Or we might be missing the reason why invalid DNS labels are allowed by IPA.

Reply:
I've opened https://github.com/podengo-project/idmsvc-api/pull/23 that is removing the pattern - so that we can choose which PR to merge.